k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1489,12 +1533,28 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
             }
         } else {
             currentMessageBuilder = null;
+
+            if (mAction == Action.REDIRECT) {
+                LocalMessage referenceMessage = mMessageReference.restoreToLocalMessage(getApplicationContext());
+                message = referenceMessage.cloneAsSuper(true);
+                message.setResentDate(new Date(), K9.hideTimeZone());
+                message.setRecipients(RecipientType.RESENT_TO, toArray(recipientPresenter.getToAddresses()));
```

Comment:
K-9 opens the connection to the mail server of the identity.
All headers (and later the body) is sent to the server.
If the server finds Resent-xxx headers then those will be used instead of the adresses in To, Cc, Bcc.
Not to the ones in To, Cc, Bcc.
Only to those in Resent-To, Resent-Cc, Resent-Bcc.

Example: Let's assume you send a message to my email1 and my email2. And i redirect from email2 to email3. Then you, email1 and email2 will not get the message again. It will only be sent to email3 and email3 will only see you as the sender and email2 and email3 as recipients.

That is by design and exactly what many people other than me seem to require as well =)